syntax = "proto3";

package mypackage;
//python -m grpc_tools.protoc -I . --python_out=. --grpc_python_out=. messaggi.proto

message Constraint {
  // Definisci i campi delle constraint
  float prezzo = 1;
  float variazione_percentuale = 2;
  float prezzo_min_24h = 3;
  float prezzo_max_24h = 4;
}

service ClientManagement {
  rpc SendData (RequestData) returns (ResponseData);
  rpc SendNewSubscriber (Subscriber) returns (ResponseData);
}

message RequestData {
  //repeated Constraint constraints = 1;
  Constraint constraints = 1;
  string topic = 2;
  string user = 3;
}

message Subscriber {
  string user = 1;
  string topic = 2;
}

message ResponseData {
  // Definisci i campi della risposta, se necessario
  string message = 1;
}